Codecademy/gamut
# fix(SelectDropdown): fix remove + remove all  buttons + medium sizing change bug for `multiple` variant

### Overview
Fixes accessibility issues:

* [[ 2.1.1] X buttons cannot be focused and not operated by the Keyboard.](https://codecademy.atlassian.net/browse/WEB-2153?atlOrigin=eyJpIjoiMzE1MTZkZWExZWUzNDg0Yjk2MTY1NDU0ZDZiNzM1NDMiLCJwIjoiaiJ9)
* [ [A] [4.1.2] Remove button is announced with [Object Object] button instead of selected member/group name.](https://codecademy.atlassian.net/browse/WEB-2154?atlOrigin=eyJpIjoiYTI5ODQwMGNlYWM0NDdjNWJlZmE1OWI1NDBmNzc1OGQiLCJwIjoiaiJ9)

Also fixes a sizing change bug on the `medium` SelectDropdown variant

### PR Checklist

- [x] Related to JIRA ticket: [WEB-2153, WEB-2154]

#### Testing Instructions

1. Navigate to SelectDropdown -> multi variant
2. Select a bunch of options
3. You should be able to navigate through the Remove X Buttons on all selected options and the Remove All X using the arrow keys, and remove them using the `Enter` button 
4. You should be able to do the same on VoiceOver (cmd + f5)
5. When items are selected + tabbing through selected items with CONTROL + OPTION + ARROW KEY, the proper "Remove OPTION" test should be read by VoiceOver


#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/33284)  | [Monolith Env](https://pr-33284-monolith.dev-eks.codecademy.com/) |
| Mono       | [Monorepo PR](https://github.com/codecademy-engineering/mono/pull/510)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-510)   |
